spec/core/responseHandling.ts

```diff
@@ -1,72 +1,67 @@
+import 'whatwg-fetch';
 import {ResponseHandler} from "../../src/ResponseHandler"
 import {GraphError} from '../../src/common'
+
 import * as mocha from 'mocha'
 
 import * as assert from 'assert';
 
-const _200_SUPERAGENT_RES = {
-  ok: true,
-  statusCode: 200,
-  body: {
-    a: 1
-  }
-};
+const _200_SUPERAGENT_RES_BODY: any = { a: 1 }; 
+const _200_SUPERAGENT_RES_INIT: ResponseInit = { status: 200 };
+const _200_SUPERAGENT_RES: Response =
+    new Response(_200_SUPERAGENT_RES_BODY, _200_SUPERAGENT_RES_INIT);
 
-const _500_SUPERAGENT_RES = {
-    ok: false,
-    response: {
-        body: {
-            error: {
-                "code": "SearchEvents",
-                "message": "The parameter $search is not currently supported on the Events resource.",
-                "innerError": {
-                    "request-id": "b31c83fd-944c-4663-aa50-5d9ceb367e19",
-                    "date": "2016-11-17T18:37:45"
-                }
-            }
+const _500_SUPERAGENT_RES_BODY: any = {
+    error: {
+        "code": "SearchEvents",
+        "message": "The parameter $search is not currently supported on the Events resource.",
+        "innerError": {
+            "request-id": "b31c83fd-944c-4663-aa50-5d9ceb367e19",
+            "date": "2016-11-17T18:37:45"
         }
-    },
-    statusCode: 500
-};
+    }
+}; 
+const _500_SUPERAGENT_RES_INIT: ResponseInit = { status: 500 };
+const _500_SUPERAGENT_RES: Response =
+    new Response(_200_SUPERAGENT_RES_BODY, _200_SUPERAGENT_RES_INIT);
 
+const error: Error = new Error();
+(error as any).statusCode = 404;
 
```

Comment:
Were you able to run the tests? I wasn't able to from your fork.  I then updated to use cross-fetch - two of the tests aren't passing. 

```
  27 passing (19ms)
  2 failing
  1) #ParseResponse() extracts code and request-id from the JSON body of 500 errors:
     TypeError: Cannot read property 'code' of null
      at lib\spec\core\responseHandling.js:38:29
      at Function.ResponseHandler.init (lib\src\ResponseHandler.js:8:13)
      at Context.<anonymous> (lib\spec\core\responseHandling.js:37:43)
  2) #ParseError() parses a 404 superagent error:
      AssertionError: -1 == 404
      + expected - actual
      --1
      +404
      
      at lib\spec\core\responseHandling.js:52:20
      at Function.ResponseHandler.init (lib\src\ResponseHandler.js:17:17)
      at Context.<anonymous> (lib\spec\core\responseHandling.js:51:43)
```